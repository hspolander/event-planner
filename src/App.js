import React from 'react';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
import * as themes from './theme';
import EventBody from './Events/components/EventBody';
import { themeType } from './types';
import './App.css';

const App = ({ theme }) => (
  <MuiThemeProvider theme={createMuiTheme(themes[theme])}>
    <EventBody />
  </MuiThemeProvider>
);

App.propTypes = {
  theme: themeType,
};

export default App;
