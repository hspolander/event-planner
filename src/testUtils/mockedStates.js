export const emptystate = {
  eventReducer: { events: null, fetching: false, error: false },
};

export const stateWithEvents = {
  eventReducer: {
    fetching: false,
    error: false,
    events: [
      {
        id: 3,
        startDate: '2019-02-26 10:00',
        endDate: '2019-02-26 12:00',
        location: 'Lagerhuset',
        activity: 'meeting',
        date: '2019-02-26 10:00:00',
      },
      {
        id: 1,
        startDate: '2019-02-26 12:00',
        endDate: '2019-02-26 13:00',
        location: 'Lagerhuset',
        activity: 'lunch',
        date: '2019-02-26 12:00:00',
      },
      {
        id: 2,
        startDate: '2019-02-26 22:00',
        endDate: '2019-02-28 07:00',
        location: 'online',
        activity: 'meeting',
        date: '2019-02-26 22:00:00',
      },
      {
        id: 2,
        startDate: '2019-02-26 22:00',
        endDate: '2019-02-28 07:00',
        location: 'online',
        activity: 'meeting',
        date: '2019-02-27 00:00:01',
      },
      {
        id: 2,
        startDate: '2019-02-26 22:00',
        endDate: '2019-02-28 07:00',
        location: 'online',
        activity: 'meeting',
        date: '2019-02-28 00:00:01',
      },
    ],
  },
};

export const stateWithError = {
  eventReducer: { events: null, fetching: false, error: true },
};

export const stateWhileLoading = {
  eventReducer: { events: null, fetching: true, error: false },
};
