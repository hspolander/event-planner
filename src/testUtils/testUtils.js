import React from 'react';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../rootReducer';

export default function renderWithRedux(
  ui,
  { initialState, store = createStore(rootReducer, initialState, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

export * from '@testing-library/react';
