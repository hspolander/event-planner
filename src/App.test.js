import React from 'react';
import App from './App';
import renderWithRedux, { getByRole, cleanup } from './testUtils/testUtils';
import { emptyState } from './testUtils/mockedStates';

describe('App', () => {
  afterEach(() => {
    cleanup();
  });
  it('renders with dark theme', () => {
    const { container } = renderWithRedux(<App />, {
      emptyState,
    });
    const rootDiv = getByRole(container, 'main');
    expect(rootDiv).toHaveStyle('background-color: rgb(29, 29, 29);');
  });

  it('should match snapshot', () => {
    const { container } = renderWithRedux(<App />, {
      emptyState,
    });
    expect(container).toMatchSnapshot();
  });
});
