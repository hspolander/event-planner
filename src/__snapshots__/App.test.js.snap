// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
    role="main"
  >
    <div
      class="MuiPaper-root makeStyles-eventListPaper-79 MuiPaper-elevation2 MuiPaper-rounded"
    >
      <ul
        aria-labelledby="events-subheader"
        class="MuiList-root MuiList-padding MuiList-subheader"
        role="list"
      >
        <div
          class="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
          id="events-subheader"
        >
          Events
        </div>
      </ul>
    </div>
    <button
      class="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiFab-label"
      >
        Toggle theme
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
